Land "Make ceil,floor,round,trunc handle integers"

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #85144

PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

lezcano committed:
Land "Make ceil,floor,round,trunc handle integers"

PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

Lezcano commented:
@JackCaoG do you mind pointing https://github.com/pytorch/xla/pull/3913 to this one? We can coordinate the landing on Monday.

@ngimel could you OK this one?

mruberry commented on torch/_torch_docs.py:
```diff
@@ -4164,6 +4167,9 @@ def merge_dicts(*dicts):
 Returns a new tensor with the floor of the elements of :attr:`input`,
 the largest integer less than or equal to each element.
 
+For integer inputs, follows the array-api convention of returning a
```
Also the convention in Python

mruberry commented:
XLA and nvFuser failures look real and see suggested doc change inline.

I suggest xfailing those tests for now and citing two issues.

mruberry reviewed (approved):
Approving for velocity but tests will need to be addressed and docs could be tweaked (see comment).

Thanks @Lezcano!

JackCaoG commented:
sg, I updated the xla pr, if all test passed I think we are good to merge.

lezcano committed:
Update on "Land "Make ceil,floor,round,trunc handle integers""


PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

lezcano committed:
Update on "Land "Make ceil,floor,round,trunc handle integers""


PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

Lezcano commented:
@JackCaoG I added the pin. See what the CI has to say.

JackCaoG commented:
@Lezcano Feel free to merge this one once all test is green, I will merge xla pr.

Lezcano commented:
@pytorchbot merge

clee2000 commented:
@pytorchbot revert -m "broke slow tests in trunk  ex https://ossci-raw-job-status.s3.amazonaws.com/log/8433956087" -c nosignal

Lezcano commented:
@janeyx99 I thought that `merge` by default would now run `-l`, so this should have been caught?

janeyx99 commented:
Oh no! Sorry for the revert--it definitely should have defaulted to -l and there seems to be a bug here! cc'ing @ZainRizvi to investigate.

lezcano committed:
Update on "Land "Make ceil,floor,round,trunc handle integers""


PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

janeyx99 commented:
@Lezcano can you try the pytorchbot merge command again when you're ready to merge something? I've merged https://github.com/pytorch/test-infra/pull/751 which I believe should fix the problem.

Lezcano commented:
will do, thanks!

lezcano committed:
Update on "Land "Make ceil,floor,round,trunc handle integers""


PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

Lezcano commented on test/test_jit_fuser_te.py:
```diff
@@ -1393,10 +1393,13 @@ def apply(fn):
-                torch.floor,
-                torch.round,
-                torch.trunc,
+                # TODO broken on int8 since
+                # https://github.com/pytorch/pytorch/pull/85144
+                # RuntimeError: Invalid integral op_type: 23
+                # torch.ceil,
+                # torch.floor,
+                # torch.round,
+                # torch.trunc,
```
Filed an issue https://github.com/pytorch/pytorch/issues/85342

lezcano committed:
Update on "Land "Make ceil,floor,round,trunc handle integers""


PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

lezcano commented:
@pytorchbot merge

lezcano committed:
Update on "Land "Make ceil,floor,round,trunc handle integers""


PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

lezcano committed:
Update on "Land "Make ceil,floor,round,trunc handle integers""


PR to land https://github.com/pytorch/pytorch/pull/78480, as Rohit does
not work in the PyTorch project anymore

[ghstack-poisoned]

lezcano commented:
@pytorchbot merge

lezcano commented:
@pytorchbot merge

lezcano commented:
@ZainRizvi there are some tests in this PR that are taking forever to run (9h+). Do you know what's going on?

lezcano commented:
@pytorchbot merge -f "windows tests are not running"

lezcano commented:
@pytorchbot merge -f "windows tests are not running"

ZainRizvi commented:
> @ZainRizvi there are some tests in this PR that are taking forever to run (9h+). Do you know what's going on?

Yeah, we're having some infra issues right now where we're severely under capacity on some machine types and it's causing tests to lag 😔
